### PR TITLE
infra: setup custom routing, fix error page, refactoring

### PR DESCRIPTION
**Pull request by LeonidShv**
1) custom routing
2) error page
3) axios interceptor
4) refactoring

### CODE REVIEW

File: error.vue [hunk 1]
```diff
@@ -1,10 +1,17 @@
...
-  <!-- TODO: test and refactor -->
   <section class="error">
     <div class="m-auto">
       <Vue3Lottie :animation-data="notFoundPage" :height="200" :width="200" />
 
-      <p class="Error-text">Not found page, please go back to the home page.</p>
+      <template v-if="error.statusCode === 404">
+        <h1>404!</h1>
+        <p class="Error-text">
+          Not found page, please go back to the home page.
```
Comment (shortSparrow) [inline error.vue:9]: If this component represents only 404 error I think better to rename the file to "notFoundPageError".
If this component is a global error, in this case better pass error code and description in props

File: lib/axios.ts [hunk 2]
```diff
@@ -1,7 +1,38 @@
...
+    emit({
+      type: "error",
+      title: "Error",
+      message: errorMessage,
+      position: "bottom-right",
+    });
+  });
+
+  errors = [];
+}, 750);
```
Comment (shortSparrow) [inline lib/axios.ts]: Looks like showNotification is a separate feature and doesn't reflect Axios, if this true, maybe should move this function into separate file and even layer

File: pages/Dashboard/components/TheDefaultChart.vue [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+  chartDefault.value = {
+    datasets,
+    labels: years,
+  };
+};
+
+onMounted(async () => {
+  await modifyDefaultChartCriteria();
+});
+</script>
```
Comment (shortSparrow) [inline pages/Dashboard/components/TheDefaultChart.vue:101]: file looks very complicated, you have refs, types, modifyDefaultChartCriteria and html. Maybe you can separate this large file into a few smaller ones

File: pages/Dashboard/composables/dashboard.ts [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    let index = 0;
+
+    for (const key in dataSetsSeries) {
+      const rest = Object.entries(dataSetsSeries[key].observations)
+        .sort((a, b) => Number(a[0]) - Number(b[0]))
+        .map((item) => Math.round(item[1][0] * tonsPerGGCoefficient));
+
+      data.push(rest[0]);
+      backgroundColor.push(colors[index] || "#607274");
+      borderColor.push(colors[index] || "#607274");
```
Comment (shortSparrow) [inline pages/Dashboard/composables/dashboard.ts]: try to create file with all colors as constants

File: pages/Dashboard/composables/dashboard.ts [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+      index++;
+    }
+    return [{ data, backgroundColor, borderColor }];
```
Comment (shortSparrow) [inline pages/Dashboard/composables/dashboard.ts:43]: why do you return array with one object?

File: error.vue (hunk 1 above)
Comment (LeonidShv) [reply error.vue:9]: This is an error page for global errors. When the application encounters some error, users are redirected to error.vue. It is implemented by Nuxt under the hood. So it is not a typical component, it is a page with auto redirects when errors are occurring 

I work with error dynamically via this composable: 
`const error = useError();`

depends on error, users can see different templates
image

File: lib/axios.ts (hunk 2 above)
Comment (LeonidShv) [reply lib/axios.ts]: move to helpers or utils

File: pages/Dashboard/components/TheDefaultChart.vue (hunk 3 above)
Comment (LeonidShv) [reply pages/Dashboard/components/TheDefaultChart.vue:101]: discusses

File: pages/Dashboard/composables/dashboard.ts (hunk 4 above)
Comment (LeonidShv) [reply pages/Dashboard/composables/dashboard.ts]: "#607274" move to constants

File: pages/Dashboard/composables/dashboard.ts (hunk 5 above)
Comment (LeonidShv) [reply pages/Dashboard/composables/dashboard.ts:43]: it is requirement from external library

File: lib/axios.ts (hunk 2 above)
Comment (LeonidShv) [reply lib/axios.ts]: 750 const

Comment (LeonidShv) [reply lib/axios.ts]: fixed

File: pages/Dashboard/composables/dashboard.ts (hunk 4 above)
Comment (LeonidShv) [reply pages/Dashboard/composables/dashboard.ts]: fixed

**Review by shortSparrow — APPROVED** [review]

**Pull request merged**